ocean_provider/utils/datatoken.py

```diff
@@ -81,19 +133,23 @@ def verify_order_tx(
             f"Multiple order events in the same transaction !!! {provider_fee_order_log}"
         )
 
+    nullify_pf_log_after_checks = False
+
     if extra_data:
         provider_data = json.loads(provider_fee_order_log.args.providerData)
         if extra_data["environment"] != provider_data["environment"]:
             raise AssertionError(
                 "Mismatch between ordered c2d environment and selected one."
             )
 
-        valid_until = provider_fee_order_log.args.validUntil
-        if (
-            datetime.utcnow().timestamp() >= valid_until
-            and not allow_expired_provider_fees
-        ):
-            raise AssertionError("Ordered c2d time was exceeded, check validUntil.")
+        time_limit = log_datetime + timedelta(
```

Comment:
can we add a predefined buffer ?
block.timestamp can be manipulated, so it's not exact

Reply:
add a env called  C2D_BUFFER_DURATION, with a default value of 120 seconds

Reply:
so time_limit = log_datetime + buffer + timedelta(....